### DISCUSSION TITLE
Discrepancy in Rainfall parameter

### DISCUSSION THREAD

**Discussion by ParameshwariS-tiger** [APSIMInitiative/ApsimX#9031] [Q&A] [answered]
Hi all, I have encountered an issue while downloading weather data from APSIM compared to the data downloaded directly from NASA POWER resource's API.  Specifically, 'rain' parameter is different across both sources while other parameters (mint, maxt, radn, wind) are same.

Below is the downloaded data from APSIM and resource's API for the same location:

APSIM - 

year  day radn  maxt   mint  rain  wind   code
()   () (MJ/m^2) (oC)  (oC)  (mm)  (m/s)    ()
2009   1   12.1  12.4  -0.8   0.0   1.9 999999
2009   2   11.1  18.9   3.8   0.0   1.3 999999
2009   3    6.5  20.5  10.2   5.1   1.1 999999
2009   4   10.6  22.4  12.7   2.3   1.2 999999
2009   5   12.7  23.1  11.9   0.0   1.4 999999
2009   6    6.6  23.1  12.0   0.0   2.6 999999
2009   7   10.3  18.7   6.8   0.0   4.0 999999
2009   8   14.4  18.7   4.2   0.0   2.8 999999
2009   9   14.7  14.8   2.0   0.0   1.1 999999
2009  10   13.4  20.4   3.4   0.0   1.6 999999

Resource's API - 

year, day, radn, maxt, mint, rain, wind
2009, 1, 12.06, 12.45, -0.83, 0.0, 1.95
2009, 2, 11.07, 18.87, 3.82, 0.27, 1.34
2009, 3, 6.51, 20.54, 10.16, 1.22, 1.08
2009, 4, 10.64, 22.4, 12.68, 0.02, 1.2
2009, 5, 12.66, 23.12, 11.86, 0.02, 1.38
2009, 6, 6.59, 23.06, 11.98, 0.73, 2.6
2009, 7, 10.35, 18.74, 6.83, 4.68, 3.98
2009, 8, 14.42, 18.73, 4.18, 0.0, 2.83
2009, 9, 14.66, 14.83, 2.01, 0.0, 1.08
2009, 10, 13.38, 20.38, 3.35, 0.08, 1.56

Could you please help me understand why there might be a difference only in the rainfall parameter?

**Comment by BrianCollinss** [comment]
The name of the dataset in the APSIM GUI is "NASA Power with Chirps Rainfall", perhaps that's why the rainfall data is different from NASA Power.

**Comment by Keith-Pembleton** [comment]
Not the issue here but CHIRPS does not cover as much of the globe as NASA POWER does.  If you use the APSIM UI outside the CHIRPS coverage it returns 0's for rainfall.  The nasapower R package has a function to create an APSIM met file.  See https://ropensci.github.io/nasapower/

**Reply by BrianCollinss** [reply, marked as answer]
Ta. This is how I cite APSIM-NextGen weather data:

> For locations outside Australia, daily temperature and radiation data were sourced from the NASA Prediction of Worldwide Energy Resources (NASA POWER) dataset (power.larc.nasa.gov), while rainfall data were obtained from the Climate Hazards Group InfraRed Precipitation with Stations (CHIRPS) dataset (Funk et al., 2014). For Australian locations, daily weather data were retrieved from the SILO database (Jeffrey et al., 2001).

**Reply by ParameshwariS-tiger** [reply]
Okay got it, thanks for the detailed clarification :)

**Reply by ParameshwariS-tiger** [reply]
Hi, Could you please guide me on where I can find the details about the process flow of creating a .met file in APSIM-NextGen using data from NASA POWER with CHIRPS dataset for a given input coordinates? Thank you!

**Reply by BrianCollinss** [reply]
You just need to right click on the weather object under Simulation and select download weather data. A window will open where you can set the parameters (coordinate, source, etc).

**Reply by ParameshwariS-tiger** [reply]
Thank you, I understood how to download weather data from APSIM but specifically, I am looking for the internal/background process like after inputting all the details how APSIM fetches data from two sources and merge them to create a met file.

**Reply by BrianCollinss** [reply]
I think you are going to need to have a look inside this file: `\APSIMNextGen\ApsimNG\Utility\WeatherDownloadDialog.cs`

**Reply by ParameshwariS-tiger** [reply]
Thanks for pointing to the reference, I'll take a look at it

**Reply by ParameshwariS-tiger** [reply]
Hi @BrianCollinss, I've reviewed the file you sent. APSIM uses three API services. I've understood the background process of two from SILO after going through their API guide. However, I couldn't find any documentation for the third API for NASA/CHIRPS: "[https://worldmodel.csiro.au/gclimate?lat={0}&lon={1}&format=apsim&start={2:yyyyMMdd}&stop={3:yyyyMMdd}](https://worldmodel.csiro.au/gclimate?lat=%7B0%7D&lon=%7B1%7D&format=apsim&start=%7B2:yyyyMMdd%7D&stop=%7B3:yyyyMMdd%7D)". Will you be able to help me with this?

**Reply by BrianCollinss** [reply]
I think your best friends are those APIs. You can find our what is done after data is retrieved from sources. I am hoping it is more than just replacing the NASA rainfall with CHIRPS data, as it would mean temperature and radiation of rainy days might not be correct.
